src/main/java/it/gov/pagopa/payhub/activities/activity/utility/SendEmailIngestionFlowActivityImpl.java

```diff
@@ -0,0 +1,125 @@
+            IngestionFlowFileDao ingestionFlowFileDao,
+            SendMailService sendMailService) {
+        this.emailTemplatesConfiguration = emailTemplatesConfiguration;
+        this.userAuthorizationService = userAuthorizationService;
+        this.ingestionFlowFileDao  = ingestionFlowFileDao;
+        this.sendMailService = sendMailService;
+    }
+
+    /**
+     * Sends an email based on the process result of the given file ingestionFlow ID.
+     *
+     * @param ingestionFlowFileId       the unique identifier of the IngestionFlow record related to the imported file.
+     * @param success      true if the process succeeded, false otherwise.
+     * @return true if the email was sent successfully, false otherwise.
+     */
+    @Override
+    public boolean sendEmail(Long ingestionFlowFileId, boolean success) {
+
+        try {
+            IngestionFlowFileDTO ingestionFlowFileDTO = ingestionFlowFileDao.findById(ingestionFlowFileId)
+                    .orElseThrow(() -> new IngestionFlowNotFoundException("Cannot found ingestionFlow having id: "+ ingestionFlowFileId));
+            String ipaCode = ingestionFlowFileDTO.getOrg().getIpaCode();
+            UserInfoDTO userInfoDTO = userAuthorizationService.getUserInfo(ipaCode, ingestionFlowFileDTO.getOperatorName());
+            UserInfoDTO organizationUserInfo = userAuthorizationService.getUserInfo(ipaCode,ipaCode+WS_USER);
+            MailTo mailTo = getMailFromIngestionFlow(ingestionFlowFileDTO, success);
+            mailTo.setTo(new String[]{userInfoDTO.getEmail()});
+            if (organizationUserInfo!= null && StringUtils.isNotBlank(organizationUserInfo.getEmail()) &&
+                ! organizationUserInfo.getEmail().equalsIgnoreCase(userInfoDTO.getEmail())) {
+                    mailTo.setCc(new String[]{organizationUserInfo.getEmail()});
+            }
+            mailTo.setHtmlText(mailTo.getHtmlText());
+            sendMailService.sendMail(mailTo);
+        }
+        catch (Exception e){
+            log.error("Sending mail failed", e);
+            return false;
+        }
+        return true;
+    }
+
+    private MailTo getMailFromIngestionFlow(IngestionFlowFileDTO ingestionFlowFileDTO, boolean success) throws Exception {
+        String template = "";
+        String flowType = ingestionFlowFileDTO.getFlowFileType();
+        if (! flowType.equalsIgnoreCase("R")) {
+            log.error("Sending e-mail not supported for flow type: {}", flowType);
+            throw new IngestionFlowTypeNotSupportedException("Sending e-mail not supported for flow type: "+flowType);
+        }
+
+        String subject = success ? emailTemplatesConfiguration.getPaymentsReportingFlowOk().getSubject()
+                : emailTemplatesConfiguration.getPaymentsReportingFlowKo().getSubject();
+        String body = success ? emailTemplatesConfiguration.getPaymentsReportingFlowOk().getBody()
+                : emailTemplatesConfiguration.getPaymentsReportingFlowKo().getBody() ;
```

Comment:
this way is more readable and less error-prone
```suggestion
        String subject = (success
                ? emailTemplatesConfiguration.getPaymentsReportingFlowOk()
                : emailTemplatesConfiguration.getPaymentsReportingFlowKo()
                ).getSubject();
        String body = (success
                ? emailTemplatesConfiguration.getPaymentsReportingFlowOk()
                : emailTemplatesConfiguration.getPaymentsReportingFlowKo()
                ).getBody() ;
```